python/ray/_private/runtime_env/conda.py

```diff
@@ -13,16 +13,42 @@
 from pathlib import Path
 
 import ray
+
+from ray.core.generated.common_pb2 import RuntimeEnv
 from ray._private.runtime_env.conda_utils import (
     get_conda_activate_commands, create_conda_env, delete_conda_env)
 from ray._private.runtime_env.context import RuntimeEnvContext
+from ray._private.runtime_env.plugin import decode_plugin_uri_from_pb
 from ray._private.utils import (get_wheel_filename, get_master_wheel_url,
                                 get_release_wheel_url, try_to_create_directory)
 from ray._private.runtime_env.packaging import Protocol, parse_uri
+from google.protobuf import json_format
 
 default_logger = logging.getLogger(__name__)
 
 
+def build_proto_conda_runtime_env(runtime_env_dict: dict,
+                                  runtime_env: RuntimeEnv):
+    """ Construct conda runtime env protobuf from runtime env dict.
+    """
+    if runtime_env_dict.get("conda"):
+        if isinstance(runtime_env_dict["conda"], str):
+            runtime_env.conda_runtime_env.conda_env_name = runtime_env_dict[
+                "conda"]
+        else:
+            runtime_env.conda_runtime_env.config = json.dumps(
+                runtime_env_dict["conda"], sort_keys=True)
+
+
+def parse_proto_conda_runtime_env(runtime_env: RuntimeEnv,
+                                  runtime_env_dict: dict):
+    """ Parse conda runtime env protobuf to runtime env dict.
+    """
+    if runtime_env.HasField("conda_runtime_env"):
```

Comment:
To make the code more Pythonic and readable, better not expose the protobuf-generated class directly here. We can have a wrapper class to encapsulate all PB-related operations.